src/promptflow/promptflow/_sdk/entities/_connection.py

```diff
@@ -627,17 +676,69 @@ class CustomConnection(_Connection):
     :type secrets: Dict[str, str]
     :param name: Connection name
     :type name: str
+    :is_custom_strong_type: Whether the custom connection is strong type.
     """
 
     TYPE = ConnectionType.CUSTOM
 
-    def __init__(self, secrets: Dict[str, str], configs: Dict[str, str] = None, **kwargs):
+    def __init__(
+        self,
+        secrets: Dict[str, str],
+        configs: Dict[str, str] = None,
+        is_azureml_custom_strong_type_connection=False,
+        **kwargs,
+    ):
+        # When create connection through file, we can't check if it is custom strong type through self.custom_type
+        # So we need a hint 'is_custom_strong_type' to indicate it.
+        if is_azureml_custom_strong_type_connection:
+            configs.update(
+                {CustomStrongTypeConnectionConfigs.FULL_TYPE: kwargs.get(CustomStrongTypeConnectionConfigs.TYPE, None)}
+            )
+            configs.update(
+                {
+                    CustomStrongTypeConnectionConfigs.FULL_MODULE: kwargs.get(
+                        CustomStrongTypeConnectionConfigs.MODULE, None
+                    )
+                }
+            )
         super().__init__(secrets=secrets, configs=configs, **kwargs)
+        self.custom_type = kwargs.get(CustomStrongTypeConnectionConfigs.TYPE, None)
 
     @classmethod
-    def _get_schema_cls(cls):
+    def _get_schema_cls(cls, is_custom_strong_type=False):
+        if is_custom_strong_type:
+            return CustomStrongTypeConnectionSchema
         return CustomConnectionSchema
 
+    @classmethod
+    def _load_from_dict(cls, data: Dict, context: Dict, additional_message: str = None, **kwargs):
+        # Note here are two cases:
+        # 1. When create/load connection data from yaml, the custom_type and module are outside the configs of data.
+        # 2. When update/load connection data from DB, the custom_type and module are within the configs of data.
+        is_custom_strong_type = data.get(CustomStrongTypeConnectionConfigs.TYPE) or (
+            data.get("configs") and data.get("configs").get(CustomStrongTypeConnectionConfigs.FULL_TYPE)
+        )
+        schema_cls = cls._get_schema_cls(is_custom_strong_type=is_custom_strong_type)
+        try:
+            loaded_data = schema_cls(context=context).load(data, **kwargs)
+        except Exception as e:
+            raise Exception(f"Load connection failed with {str(e)}. f{(additional_message or '')}.")
+        return cls(
+            base_path=context[BASE_PATH_CONTEXT_KEY],
+            is_azureml_custom_strong_type_connection=is_custom_strong_type,
+            **loaded_data,
+        )
+
+    def __setattr__(self, key, value):
+        if hasattr(self, "custom_type") and self.is_custom_strong_type():
+            if isinstance(value, Secret) and hasattr(self, "secrets") and key in self.secrets:
```

Comment:
We'd like to deprecate the `Secret` class usage, and do not override the `__setattr__`, as there are some limitations if we support `connection.key=xx`, for example the 'name', 'module', all the class members can't be used in custom connection, and will hit odd errors, we could sync about this offline.

Reply:
removed setattr